Source/ARTHttp.m

```diff
@@ -175,18 +179,19 @@ - (void)executeRequest:(NSMutableURLRequest *)request completion:(void (^)(NSHTT
     [self.logger debug:@"%@ %@", request.HTTPMethod, request.URL.absoluteString];
     [self.logger verbose:@"Headers %@", request.allHTTPHeaderFields];
 
+    __weak typeof(self) weakSelf = self;
```

Comment:
OK, but you just said that Realtime will live until `close()`  is called. You mean called internally then? I thought you meant called manually by the user, which didn't make sense to me, as obviously the user still has a reference afterwards, namely the one it just called `closed()` on!

Anyway, if that's what's happening then yes, sounds good.